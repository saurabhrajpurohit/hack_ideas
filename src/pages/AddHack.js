import React, { useEffect } from 'react';
import { Stack, Box, LinearProgress, Alert, Container, Typography } from '@mui/material';
import { HackathonService, TagsService } from '../service/Service';
import HackForm from '../components/HackForm';
import { useNavigate } from 'react-router';

export default function AddHack() {
    const navigate = useNavigate();
    const [loading, setLoading] = React.useState(false);
    const [tags, setTags] = React.useState([]);
    const [error, setError] = React.useState("");

    useEffect(() => {
        getTags();
    }, []);

    function getTags() {
        TagsService.getAll()
            .then(data => setTags(data))
            .catch(err => {
                console.error(err);
            });;
    };

    function handleSubmit(add, hackathon) {
        if (add) {
            setLoading(true);
            HackathonService.create(hackathon).then(data => {
                navigate("/hack/list");
                setLoading(false);
            }).catch(err => {
                setLoading(false);
                console.error(err);
                setError("We are unable to create the hackathon at this time. Please try again later.");
            });
        } else {
            navigate("/hack/list");
        }
    };

    return (
        <Container maxWidth="sm">
            <Stack sx={{ mb: 5 }}>
                <Typography variant="h4" gutterBottom>
                    Add Hackathon
                </Typography>
                <Typography sx={{ color: 'text.secondary' }}>Please add the below details to add a new hackathon.</Typography>
            </Stack>
            {loading && <Box sx={{ width: '100%' }} mb={1}>
                <LinearProgress />
            </Box>}
            {error && <Box sx={{ width: '100%' }} pb={1}>
                <Alert severity="error">{error}</Alert>
            </Box>}
            <HackForm handleSubmit={handleSubmit} tags={tags} />
        </Container>
    )
}