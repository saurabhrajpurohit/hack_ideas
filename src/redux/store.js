import { createStore } from "redux";
import { SET_USER } from "../utils/constants";

const userState = {};

const userReducer = (state = userState, action) => {
    switch (action.type) {
        case SET_USER:
            return {
                ...state,
                user: action.value
            }
        default:
            return state;
    }
};

export default createStore(userReducer);

