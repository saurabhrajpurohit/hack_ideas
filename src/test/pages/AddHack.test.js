import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import AddHack from '../../pages/AddHack';
import store from '../../redux/store';

test('ui to be rendered correctly', () => {
    render(
        <Provider store={store}>
            <BrowserRouter>
                <AddHack />
            </BrowserRouter>
        </Provider>
    );
    const element = screen.getByTestId('add-hack-page');
    expect(element).toMatchSnapshot();
});
