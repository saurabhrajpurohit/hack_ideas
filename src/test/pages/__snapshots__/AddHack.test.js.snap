// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui to be rendered correctly 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  data-testid="add-hack-page"
>
  <div
    class="css-lr1r4n-MuiStack-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
    >
      Add Hackathon
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
    >
      Please add the below details to add a new hackathon.
    </p>
  </div>
  
  <div
    class="css-nx5yqs-MuiStack-root"
    data-testid="hack-add-form"
  >
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="title"
        id="title-label"
      >
        Title
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="title"
          name="Title"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-173wfxe"
          >
            <span>
              Title
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="description"
        id="description-label"
      >
        Description
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <textarea
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          id="description"
          name="Description"
          rows="4"
          style="height: 0px; overflow: hidden;"
        />
        <textarea
          aria-hidden="true"
          class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-173wfxe"
          >
            <span>
              Description
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="description"
        id="description-label"
      >
        Event Date
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="description"
          name="EventDate"
          type="date"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1z7n62"
          >
            <span>
              Event Date
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-e363es-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        id="hackathon-tags-label"
      >
        Tags
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="hackathon-tags-label tags"
          class="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="tags"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          id="select-tags"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-173wfxe"
          >
            <span>
              Tags
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
      />
    </div>
    <div
      class="css-gmwslw-MuiStack-root"
    >
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Cancel
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        data-testid="hack-form-submit"
        tabindex="0"
        type="button"
      >
        Save
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
